ui/apps/platform/src/Containers/Collections/CollectionAttacher.tsx

```diff
@@ -1,7 +1,95 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
+import { Alert, Button, debounce, Flex, SearchInput, Truncate } from '@patternfly/react-core';
 
-function CollectionAttacher() {
-    return <></>;
+import BacklogListSelector from 'Components/PatternFly/BacklogListSelector';
+import { CollectionResponse } from 'services/CollectionsService';
+import useEmbeddedCollections from './hooks/useEmbeddedCollections';
+
+const selectorListCells = [
+    {
+        name: 'Name',
+        render: ({ name }) => (
+            <Button variant="link" className="pf-u-pl-0" isInline>
+                {name}
+            </Button>
+        ),
+    },
+    {
+        name: 'Description',
+        render: ({ description }) => <Truncate content={description} />,
+    },
+];
+
+export type CollectionAttacherProps = {
+    initialEmbeddedCollections: CollectionResponse[];
+    onSelectionChange: (collections: CollectionResponse[]) => void;
+};
+
+function compareNameLowercase(search: string): (item: { name: string }) => boolean {
+    return ({ name }) => name.toLowerCase().includes(search.toLowerCase());
+}
+
+function CollectionAttacher({
+    initialEmbeddedCollections,
+    onSelectionChange,
+}: CollectionAttacherProps) {
+    const [search, setSearch] = useState('');
+    const embedded = useEmbeddedCollections(initialEmbeddedCollections);
+    const { attached, detached, attach, detach, hasMore, fetchMore, onSearch } = embedded;
+    const { isFetchingMore, fetchMoreError } = embedded;
+
+    const onSearchInputChange = useMemo(
+        () =>
+            debounce((value: string) => {
+                setSearch(value);
+                onSearch(value);
+            }, 800),
+        [onSearch]
+    );
+
+    const selectedOptions = attached.filter(compareNameLowercase(search));
+    const deselectedOptions = detached.filter(compareNameLowercase(search));
+
+    return (
+        <Flex direction={{ default: 'column' }} spaceItems={{ default: 'spaceItemsXl' }}>
+            <SearchInput
+                aria-label="Search by name"
+                placeholder="Search by name"
```

Comment:
Although I see **Search** Google and Gmail, **Filter** seems more common for placeholder in UI.

Reply:
Fair, there are multiple occurrences of **Filter** in similar contexts but this looks to be the first case where we use **Search**. This is kind of a hybrid case, as we are both searching and filtering the two lists with a single input, but I'll change for the sake of consistency.